v2v-helper/migrate/migrate.go

```diff
@@ -61,26 +62,69 @@ func (migobj *Migrate) AddVolumestoHost(vminfo vm.VMInfo) (vm.VMInfo, error) {
 		}
 	}
 	migobj.logMessage("Volumes created successfully")
+	return vminfo, nil
+}
 
+func (migobj *Migrate) AttachVolume(disk vm.VMDisk) (string, error) {
+	openstackops := migobj.Openstackclients
 	migobj.logMessage("Attaching volumes to VM")
+
+	err := openstackops.AttachVolumeToVM(disk.OpenstackVol.ID)
+	if err != nil {
+		return "", fmt.Errorf("failed to attach volume to VM: %s", err)
+	}
+
+	// Get the Path of the attached volume
+	devicePath, err := openstackops.FindDevice(disk.OpenstackVol.ID)
+	if err != nil {
+		return "", fmt.Errorf("failed to find device: %s", err)
+	}
+	return devicePath, nil
+}
+
+func (migobj *Migrate) DetachVolume(disk vm.VMDisk) error {
+	openstackops := migobj.Openstackclients
+	err := openstackops.DetachVolumeFromVM(disk.OpenstackVol.ID)
+	if err != nil {
+		return fmt.Errorf("failed to detach volume from VM: %s", err)
+	}
+	err = openstackops.WaitForVolume(disk.OpenstackVol.ID)
+	if err != nil {
+		return fmt.Errorf("failed to wait for volume to become available: %s", err)
+	}
+	return nil
+}
+
+func (migobj *Migrate) DetachAllVolumes(vminfo vm.VMInfo) error {
+	openstackops := migobj.Openstackclients
 	for _, vmdisk := range vminfo.VMDisks {
-		err := openstackops.AttachVolumeToVM(vmdisk.OpenstackVol.ID)
+		err := openstackops.DetachVolumeFromVM(vmdisk.OpenstackVol.ID)
 		if err != nil {
-			return vminfo, fmt.Errorf("failed to attach volume to VM: %s", err)
+			if strings.Contains(err.Error(), "is not attached to volume") {
+				return nil
+			}
+			return fmt.Errorf("failed to detach volume from VM: %s", err)
+		}
+		err = openstackops.WaitForVolume(vmdisk.OpenstackVol.ID)
+		if err != nil {
+			return fmt.Errorf("failed to wait for volume to become available: %s", err)
 		}
-		migobj.logMessage(fmt.Sprintf("Volume attached to VM: %s", vmdisk.OpenstackVol.Name))
+		log.Printf("Volume %s detached from VM\n", vmdisk.Name)
 	}
+	time.Sleep(1 * time.Second)
+	return nil
+}
 
-	// Get the Path of the attached volume
-	for idx, vmdisk := range vminfo.VMDisks {
-		devicePath, err := openstackops.FindDevice(vmdisk.OpenstackVol.ID)
+func (migobj *Migrate) DeleteAllDisks(vminfo vm.VMInfo) error {
```

Comment:
For the sake of consistency, all function calls for operations on Volumes should probably be named accordingly. I think this should be: `DeleteAllVolumes`.